docs/design/free_credit_offer_typesDesignDoc.md

```diff
@@ -0,0 +1,253 @@
+  Event:  "sign-ups",
+  Properties: analytics.NewProperties().Set("referred", "true"),
+})
+```
+
+How to send an event from Satellite GUI to customer.io?
+
+```javascript
+analytics.track('Signed Up', {
+  referred_by: ‘brandonisawesome’
+});
+```
+
+**List of trigger events**:
+- Send_referral event - triggered from the Satellite GUI
+- Referral_redeemed event - triggered from the consoleweb server.
+Reason: activation success page is served as a static page, therefore we need to send this event from the back-end after user activated their account
+
+How to send notification for a specific use based on their state?
+
+By utilizing customer.io’s attribute feature, we can update user’s state through each step in our referral program and deliver appropriate content to them.
+
+**How to update user’s state in customer.io?**
+
+```javascript
+analytics.identify("97980cfea0067", {
+  name: "Peter Gibbons",
+  email: "peter@initech.com",
+  low_credits: true,
+});
+```
+
+Then,we will be using Customer.io’s segment triggered campaign to send out notifications.
+
+How does email get sent out when admin starts a new offer?
+Due to the limitation of the customer.io API, we can’t create a new campaign through an API call. An Admin needs to go to customer.io web interface to create a new campaign manually.
+
+### Front End
+
+### Satellite GUI
+
+**src/store/modules/userReferralStats.ts**
+- A store for managing user’s referral stats from the backend
+
+**src/components/referral/***.ts**
+- Based on mockups, we will create components to display user’s referral stats
+
+**src/components/account/AccountArea.vue**
+- Add display for user’s referral link and copy to clipboard button using vue-clipboard2
+
+**github.com/storj/tardigrade-satellite-theme**
+- Add `.env` file that stores segment tracking id for each satellite 
+
+### Admin GUI
+
+we will be using go template for the UI
+**/web/admin/offer/home.html**
+
+## Rationale
+
+As if the current user_credit table design, we will have a new entry each time when a user earns a credit. The reason why we designed this way is due to the starting date for the expiration date of credits. Each credit will be expired at a different time based on the duration we set for a particular offer and the date the credit is awarded to a user.
+
+We will check against the credit duration interval when inserting a new row into the user_credit table and update the expires_at for each entry accordingly.
+
+## Open issues (if applicable)
+
+1. We will have other marketing programs, for example, open source partner program. Should we create a top level system for all the marketing related services since we will probably be using the same private port and admin GUI for the marketing team to manage configurations for all the programs? (marketing service)
+2. sync up with the Matt and Egon about how are we going to restrict ppl from accessing the private port? Only allow local access or restricting people through firewall?
```

Comment:
The least developer time would be to only listen on localhost and then setup a reverse-proxy with password authentication (or similar).

Reply:
It sounded like Matt asked for something listening on an external interface, which could then be limited via firewall to the vpn. So there is some sort of reconciliation to do